web_generator/lib/src/translator.dart

```diff
@@ -732,34 +714,41 @@ class Translator {
                 'Unexpected interfacelike type ${interfacelike.type}');
         }
       }
+    }
+  }
+
+  /// Given an [interfacelikeName], combines its interfacelike declaration, its
+  /// partial interfacelikes, and any mixins it includes in that order.
+  ///
+  /// Mixins are applied by applying the mixin interface first and then its
+  /// partial interfaces.
+  void _combineInterfacelikes(String interfacelikeName) {
+    final decl = _typeToDeclaration[interfacelikeName]! as idl.Interfacelike;
+    for (final interfacelike in [
+      decl,
+      ..._typeToPartials[interfacelikeName] ?? <idl.Interfacelike>[]
+    ]) {
+      _addOrUpdateInterfaceLike(interfacelike);
+    }
+    final mixins = _includes[interfacelikeName];
+    if (mixins == null) return;
+    for (final mixin in mixins) {
       for (final interfacelike in [
-        ...library.interfaceMixins,
-        ...library.partialInterfaceMixins
+        _typeToDeclaration[mixin] as idl.Interfacelike,
+        ..._typeToPartials[mixin] ?? <idl.Interfacelike>[]
```

Comment:
nit: use `...?_typeToPartials[mixin]` instead of `??`?

Reply:
Ooh, I don't think I knew this existed. Thanks!